code/modules/clothing/spacesuits/rig/modules/ds13/healthbar.dm

```diff
@@ -21,27 +21,40 @@
 
 	base_type = /obj/item/rig_module/healthbar
 
-/obj/item/rig_module/healthbar/proc/register_user(var/mob/newuser)
-	user = newuser
-	GLOB.updatehealth_event.register(user, src, /obj/item/rig_module/healthbar/proc/update)
-	GLOB.death_event.register(user, src, /obj/item/rig_module/healthbar/proc/death)
+	var/tracking_level = RIG_SENSOR_OFF
+	var/tracking_mode = RIG_SENSOR_MANUAL
 
-/obj/item/rig_module/healthbar/proc/unregister_user()
-	GLOB.updatehealth_event.unregister(user, src, /obj/item/rig_module/healthbar/proc/update)
-	GLOB.death_event.unregister(user, src, /obj/item/rig_module/healthbar/proc/death)
-	user = null
+/obj/item/rig_module/healthbar/Initialize()
+	. = ..()
+	tracking_mode = pick(RIG_SENSOR_MANUAL, RIG_SENSOR_AUTOMATIC)
+
+	if(tracking_mode == RIG_SENSOR_AUTOMATIC)
+		GLOB.vitals_auto_update_tracking += src
+		automatic_tracking_update()
+
+	else
+		tracking_level = pick(RIG_SENSOR_OFF, RIG_SENSOR_BINARY, RIG_SENSOR_VITAL, RIG_SENSOR_TRACKING)
 
 /obj/item/rig_module/healthbar/Destroy()
 	unregister_user()
+	GLOB.vitals_auto_update_tracking -= src
 	.=..()
 
-
 /obj/item/rig_module/healthbar/rig_equipped(var/mob/user, var/slot)
 	register_user(user)
 
 /obj/item/rig_module/healthbar/rig_unequipped(var/mob/user, var/slot)
 	unregister_user()
 
+/obj/item/rig_module/healthbar/proc/register_user(var/mob/newuser)
```

Comment:
Add a rig var to hold the healthbar, set and unset it in these procs

Reply:
Done